packages/ckeditor5-package-generator/lib/templates/js/eslint.config.js

```diff
@@ -0,0 +1,78 @@
+import globals from 'globals';
+import { defineConfig } from 'eslint/config';
+import eslintConfigCKEditor5 from 'eslint-config-ckeditor5';
+import eslintPluginCKEditor5Rules from 'eslint-plugin-ckeditor5-rules';
+
+export default defineConfig( [
+	eslintConfigCKEditor5,
+	{
+		name: 'Ignored files config',
+		ignores: [
+			// Ignore the entire `dist/`.
+			'dist/**'
+		]
+	},
+	{
+		name: 'Base config',
+		languageOptions: {
+			ecmaVersion: 'latest',
+			sourceType: 'module'
+		},
+		linterOptions: {
+			reportUnusedInlineConfigs: 'error',
+			reportUnusedDisableDirectives: 'error'
+		},
+		plugins: {
+			'ckeditor5-rules': eslintPluginCKEditor5Rules
+		},
+		rules: {
+			// This rule disallows importing from any path other than the package main entrypoint.
+			'ckeditor5-rules/allow-imports-only-from-main-package-entry-point': 'error',
+			// As required by the ECMAScript (ESM) standard, all imports must include a file extension.
+			// If the import does not include it, this rule will try to automatically detect the correct file extension.
+			'ckeditor5-rules/require-file-extensions-in-imports': [ 'error', {
+				extensions: [ '.ts', '.js', '.json' ]
+			} ],
+			// This rule ensures that all imports from `@ckeditor/*` packages are done through the main package entry points.
+			// This is required for the editor types to work properly and to ease migration to the installation methods
+			// introduced in CKEditor 5 version 42.0.0.
+			'ckeditor5-rules/no-legacy-imports': 'error'
+		}
+	},
+	{
+		name: 'NodeJS environment config',
+		files: [
+			'./scripts/**/*.mjs'
```

Comment:
```suggestion
			'./scripts/**/*.@(js|mjs|cjs)'
```

If a package is a module (`type=module`), the default extension is `js` instead of `mjs`. However, creating a CommonJS script (`cjs`) is still possible. To avoid _wtf feelings_, let's cover all cases.

Please, update the previous outputs, too. TypeScript accepts `cts` as well.